### PR TITLE
50142_Gem_JS_CSS_Config_layouts

### PR DESCRIPTION
**Pull request by noor-framgia**

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,14 +1,25 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <title>Brs19</title>
-  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <title><%= t :home_head %></title>
+  <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track" => true %>
```
Comment (abdullahallmehedi) [inline app/views/layouts/application.html.erb]: 80

**Comment by SSalekin** [comment]
ok.

**Comment by abdullahallmehedi** [comment]
ok

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,4 +2,13 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :set_locale
+
+  def set_locale
```
Comment (MuhammadTamzid) [inline app/controllers/application_controller.rb]: Private

**Comment by manazirahsan** [comment]
okey

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
...
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :set_locale
+
+  private
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
+
+  def default_url_options(options = {})
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: remove `()`

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,14 +1,26 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <title>Brs19</title>
-  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <title><%= t :home_head %></title>
+  <%= stylesheet_link_tag "application", media: "all",
+                          "data-turbolinks-track" => true %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: `"data-turbolinks-track": true`

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,14 +1,26 @@
...
+  <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: `"data-turbolinks-track": true`

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -1,14 +1,26 @@
...
-
-<%= yield %>
-
+<%= render :header %>
+<div class="container">
+  <div id="nav">
+    <%= t :language %>:
+    <%= link_to_unless_current t(:english), locale: "en" %>
+  </div>
+  <% flash.each do |message_type, message| %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: should split it to new file

File: config/application.rb [hunk 7]
```diff
@@ -16,11 +16,12 @@ class Application < Rails::Application
     # Run "rake -D time" for a list of tasks for finding time zone names. Default is UTC.
     # config.time_zone = 'Central Time (US & Canada)'
 
-    # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
-    # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
-    # config.i18n.default_locale = :de
+    config.i18n.enforce_available_locales = false
+    config.i18n.available_locales = [:en]
```
Comment (dieunb) [inline config/application.rb]: should define more language, example: `japanese`

File: app/views/layouts/application.html.erb (hunk 6 above)
Comment (noor-framgia) [inline app/views/layouts/application.html.erb]: write in new file and render here?

Comment (dieunb) [inline app/views/layouts/application.html.erb]: yep

**Pull request merged**